src/lib/ccf/index.ts

```diff
@@ -0,0 +1,458 @@
+import {INSTANCE_TYPE_COMPUTE_PROCESSOR_MAPPING} from '@cloud-carbon-footprint/aws/dist/lib/AWSInstanceTypes';
+import {IImpactModelInterface} from '../interfaces';
+import Spline from 'typescript-cubic-spline';
+import * as aws_instances from './aws-instances.json';
+import * as gcp_instances from './gcp-instances.json';
+import * as gcp_use from './gcp-use.json';
+import * as aws_use from './aws-use.json';
+import * as azure_use from './azure-use.json';
+import * as azure_instances from './azure-instances.json';
+import * as gcp_embodied from './gcp-embodied.json';
+import * as aws_embodied from './aws-embodied.json';
+import * as azure_embodied from './azure-embodied.json';
+import {KeyValuePair} from '../../types/boavizta';
+
+// consumption information for a single instance
+interface IConsumption {
+  idle?: number;
+  tenPercent?: number;
+  fiftyPercent?: number;
+  hundredPercent?: number;
+  minWatts?: number;
+  maxWatts?: number;
+}
+
+// information about a single compute instance
+interface IComputeInstance {
+  consumption: IConsumption;
+  embodiedEmission?: number;
+  name: string;
+  vCPUs?: number;
+  maxVCPUs?: number;
+}
+
+export enum Interpolation {
+  LINEAR = 'linear',
+  SPLINE = 'spline',
+}
+
+export interface ICcfResult {
+  e: number;
+  m: number;
+}
+
+export class CloudCarbonFootprint implements IImpactModelInterface {
+  // Defined for compatibility. Not used in CCF.
+  authParams: object | undefined;
+  // name of the data source
+  name: string | undefined;
+  // compute instances grouped by the provider with usage data
+  private computeInstances: {
+    [key: string]: {
+      [key: string]: IComputeInstance;
+    };
+  } = {};
+
+  // list of all the compute instances by Architecture
+  private gcpList: KeyValuePair = {};
+  private azureList: KeyValuePair = {};
+  private awsList: KeyValuePair = {};
+
+  private provider = '';
+  private instanceType = '';
+  private expectedLifespan = 4;
+
+  private interpolation = Interpolation.LINEAR;
+
+  constructor() {
+    this.standardizeInstanceMetrics();
+  }
+
+  /**
+   * Defined for compatibility. Not used in CCF.
+   */
+  authenticate(authParams: object): void {
+    this.authParams = authParams;
+  }
+
+  /**
+   *  Configures the CCF Plugin for IEF
+   *  @param {string} name name of the resource
+   *  @param {Object} staticParams static parameters for the resource
+   *  @param {("aws"|"gcp"|"azure")} staticParams.provider aws, gcp, azure
+   *  @param {string} staticParams.instance_type instance type from the list of supported instances
+   *  @param {number} staticParams.expected_lifespan expected lifespan of the instance in years
+   *  @param {Interpolation} staticParams.interpolation linear(All Clouds), spline (only for AWS)
```

Comment:
I think since we are using typescript types, we can skip param types and just describe them to not have conflict.
F. ex. in this case you have described the `staticParams` object in the doc, but its typescript type is `object` so it's better to focus on typescript types.

Reply:
it is kept that way to avoid issues when compiling to different languages in JSII